File: swiftui/swiftui_prototype/swiftui_prototype/Models/BTDevice.swift
//
//  BTDevice.swift
//  swiftui_prototype
//
//  Created by Timo van der Haar on 15/09/2022.
//

import Foundation

struct BTDevice: Identifiable {
    var id: UUID
    var name: String
    var batteryPercentage: Int
    var manufacturer: String
    
    init(id: UUID = UUID(), name: String, batterPercentage: Int, manufacturer: String) {
        self.id = id
        self.name = name
        self.batteryPercentage = batterPercentage
        self.manufacturer = manufacturer
    }
}

extension BTDevice {
    static let sampleData: [BTDevice] =
    [
        BTDevice(name: "Device A", batterPercentage: 50, manufacturer: "China"),
        BTDevice(name: "Device B", batterPercentage: 80, manufacturer: "China"),
        BTDevice(name: "Device C", batterPercentage: 1, manufacturer: "China"),
        BTDevice(name: "Device D", batterPercentage: 20, manufacturer: "China")
    ]
}

File: kotlin-multiplatform-mobile/KotlinMultiplatformPrototype/shared/src/commonMain/kotlin/com/example/kotlinmultiplatformprototye/BTDevice.kt
package com.example.kotlinmultiplatformprototye

//class BTDevice {
//    var id: String = ""
//    var name: String = ""
//    var batteryPercentage: Int = 0
//    var manufacturer: String = ""
//}

data class BTDevice(val id: String, val name: String, val batteryPercentage: Int, val manufacturer: String) {
    companion object {
        fun sampleData() = listOf(
        BTDevice("abcd", "Device A", 80, "China"),
        BTDevice("efgh", "Device B", 20,"China"),
        BTDevice("ijkl", "Device A", 1, "China"),
        BTDevice("mnop", "Device A", 69, "China")
        )
    }
}